### PR TITLE
Add calls for filtering report severity list

### PR DESCRIPTION
**Pull request by lilian-houdelet**
When we merge this PR we have to invalidate all nodes on dev-opf in order to delete all existing logs. Otherwise, logs wont be correctly displayed (cf invalidate_nodes_builds.py from admin-tools)

### CODE REVIEW

File: src/components/report-viewer/report-viewer.js [hunk 1]
```diff
@@ -199,8 +219,12 @@ export default function ReportViewer({
         setSelectedNode(rootId);
         setExpandedNodes([rootId]);
         setLogs(rootReport.current.getAllLogs());
-        setSelectedSeverity(LogReportItem.getDefaultSeverityFilter());
-    }, [jsonReportTree, createReporterItem]);
+        Promise.resolve(getSeverityPromise(rootId)).then((response) => {
```
Comment (flomillot) [inline src/components/report-viewer/report-viewer.js:222]: Why `Promise.resolve `?
Can you catch and log the eventual error ?

File: src/components/report-viewer/report-viewer.js [hunk 2]
```diff
@@ -220,7 +244,12 @@ export default function ReportViewer({
 
     const selectNode = (nodeId) => {
         if (selectedNode !== nodeId) {
-            refreshNode(nodeId, selectedSeverity);
+            Promise.resolve(getSeverityPromise(nodeId)).then((fetchedData) => {
```
Comment (flomillot) [inline src/components/report-viewer/report-viewer.js:247]: Same

File: src/components/results/common/computation-report-viewer.tsx [hunk 3]
```diff
@@ -107,13 +109,32 @@ export const ComputationReportViewer: FunctionComponent<
...
 
+    const subReportSeveritiesPromise = (reportId: string) => {
+        fetchSubReportSeverity(
+            studyUuid.toString(),
+            currentNode.id.toString(),
+            reportId
+        );
+    };
+
+    const reportSeveritiesPromise = (reportId: string) => {
```
[the remark this answers is not in the document]
Comment (lilian-houdelet) [reply src/components/results/common/computation-report-viewer.tsx:120]: I used the same parameters than the one that gather the reports

**Review by dbraquart — CHANGES_REQUESTED** [review]
As discussed, severity list should be returned as a taskValue per LogReport (tree node on the left pane). No REST call should be needed.

File: src/components/report-viewer/log-report.js [hunk 4]
```diff
@@ -80,6 +82,14 @@ export default class LogReport {
         return this.logs;
     }
 
+    getSeverityList() {
+        return this.severityList;
+    }
+
+    getAllSeverityList() {
```
Comment (flomillot) [inline src/components/report-viewer/log-report.js:89]:
```suggestion
    // Represent all the different severities of this report and it's subreports
    getAllSeverityList() {
```
I would add something like that

File: src/components/report-viewer/log-report-item.js [hunk 5]
```diff
@@ -60,11 +60,19 @@ export default class LogReportItem {
         },
     };
 
-    static getDefaultSeverityFilter() {
+    static getDefaultSeverityFilter(severityList) {
         const filterConfig = {};
-        Object.values(LogReportItem.SEVERITY).forEach((s) => {
-            filterConfig[s.name] = s.displayedByDefault;
-        });
+        if (!severityList) {
```
Comment (flomillot) [inline src/components/report-viewer/log-report-item.js:65]:
```suggestion
        if (!severityList && !severityList.lengh) {
```

File: src/components/report-viewer/log-report.js (hunk 4 above)
Comment (lilian-houdelet) [reply src/components/report-viewer/log-report.js:89]: I added it in the constructor and not on the getter

**Review by flomillot — APPROVED** [review]
Code OK

**Review by dbraquart — CHANGES_REQUESTED** [review]
code: ok
tests: ok

But we still have an issue: how do we treat the former logs compatibility ? to be discussed with TLs

**Review by dbraquart — APPROVED** [review]
code: ok
tests: ok

**Pull request merged**